app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -771,12 +774,23 @@ public void onVisibilityChanged(@NonNull Status.Visibility visibility) {
         setStatusVisibility(visibility);
     }
 
-    private void updateVisibleCharactersLeft() {
-        int charactersLeft = maximumTootCharacters - textEditor.length();
+    int calculateRemainingCharacters() {
+        int offset = 0;
+        URLSpan[] urlSpans = textEditor.getUrls();
+        if (urlSpans != null) {
+            for (URLSpan span : urlSpans) {
+                offset += Math.max(0, span.getURL().length() - MAXIMUM_URL_LENGTH);
```

Comment:
Sorry, I don't get it - you subtract maximum toot length from actual link length?..

Reply:
`MAXIMUM_URL_LENGTH` is the length that mastodon uses (23)